src/utilities/language/cache-visit-preferred-language.ts

```diff
@@ -2,10 +2,8 @@ import { navigate } from 'gatsby';
 import { PREFERRED_LANGUAGE_KEY, safeWindow } from 'src/utilities';
 
 export const cacheVisitPreferredLanguage = (isPageLanguageDefault: boolean, language: string, href: string) => {
-  if (isPageLanguageDefault) {
-    safeWindow.localStorage.clear();
-  } else {
-    safeWindow.localStorage.setItem(PREFERRED_LANGUAGE_KEY, language);
+  if (!isPageLanguageDefault) {
+    safeWindow.sessionStorage.setItem(PREFERRED_LANGUAGE_KEY, language);
```

Comment:
Could this be handled by the context instead? Or by a dedicated function in the context file, again just to hide the implementation details of how it all works